mcos/optimizer.py

```diff
@@ -173,7 +173,7 @@ def _get_optimal_portfolio(self, cov: np.array, mu: np.array) -> np.array:
         :param mu: vector of expected returns
         :return: optimal portfolio allocation
         """
-        inv = np.linalg.inv(cov)
+        inv = np.linalg.pinv(cov)
```

Comment:
Occasionally we end up with a singular matrix which raises an exception when trying to compute the inverse. Since it's impossible to get the inverse of a singular matrix, I changed this to get the pseudo inverse instead.

Reply:
Ok. Can you do try `inv` first, and fall back to `pinv` if that fails?

Reply:
For sure!